app/views/settings/edit.html.erb

```diff
@@ -6,11 +6,18 @@
     <p><%= t(".mfa.enabled_html") %></p>
     <%= form_tag multifactor_auth_path, method: :put, id: "mfa-edit" do %>
       <%= label_tag :level, t(".mfa.level.title"), class: "form__label" %>
-      <%= select_tag :level, options_for_select([
-        [t(".mfa.level.ui_and_api"), "ui_and_api"],
-        [t(".mfa.level.ui_and_gem_signin"), "ui_and_gem_signin"],
-        [t(".mfa.level.ui_only"), "ui_only"],
-        [t(".mfa.level.disabled"), "disabled"]], @user.mfa_level), class: "form__input form__select" %>
+
+      <% mfa_options = [
+          [t(".mfa.level.ui_and_api"), "ui_and_api"],
+          [t(".mfa.level.ui_and_gem_signin"), "ui_and_gem_signin"],
+          [t(".mfa.level.disabled"), "disabled"]] %>
+
+      <% if @user.mfa_ui_only? %>
+        <% mfa_options.insert(2, [t(".mfa.level.ui_only"), "ui_only"]) %>
+      <% end %>
```

Comment:
It's best practice to not define variables like this in the view, but instead to set them in the [controller](https://github.com/Shopify/rubygems.org/blob/ca2ed5d6a47c0adf81cc541c63aa3bf90d855199/app/controllers/settings_controller.rb#L5) as an instance variable.